### ISSUE TITLE
Allow URL fragments in redirect rules

### ISSUE THREAD

**Issue by dboulet** [gohugoio/hugo#13359]
I ran into an issue when I added the following redirect rule to `config/development/server.yml`:
```yaml
redirects:
- force: false
  from: '/accessibility/accessibility_plan/'
  status: 301
  to: '/accessibility/#plan'
```
I received the following error in the console:
```
Error: command error: failed to create config from result: failed to decode "server": unsupported redirect to value "/accessibility/#plan" in server config; currently this must be either a remote destination or a local folder, e.g. "/blog/" or "/blog/index.html"
```
It seems like it should be valid to include a URL fragment within the value for `to` in a redirect rule.

Could the rule validation in Hugo be modified to allow fragments?

**Comment by bep** [comment]
>Could the rule validation in Hugo be modified to allow fragments?

I revamped all of this in 029d1e0ced541450fe7f056eaa438a1f6ac28685 a few days ago (not released, I will do a release sometime next week, I think), and I'm pretty sure HEAD version of Hugo supports fragments in `Redirect.To`.

**Comment by dboulet** [comment]
Thanks, @bep. I can confirm that the error above doesn’t appear when using the HEAD version of Hugo.
